test/common/crudtester.go

```diff
@@ -412,11 +422,20 @@ func (c *FederatedTypeCrudTester) waitForPropagatedVersion(template, placement,
 
 	clusterNames := c.adapter.Placement().ClusterNames(placement)
 	selectedClusters := sets.NewString(clusterNames...)
+	if c.kind == federatedtypes.NamespaceKind {
+		// Delete the primary cluster as it will never be included in
+		// PropagatedVersion's list of cluster verions.
+		selectedClusters.Delete(c.getPrimaryClusterName())
+	}
 
 	var version *fedv1a1.PropagatedVersion
 	err := wait.PollImmediate(c.waitInterval, c.clusterWaitTimeout, func() (bool, error) {
 		var err error
-		version, err = client.FederationV1alpha1().PropagatedVersions(namespace).Get(name, metav1.GetOptions{})
+		if c.kind == federatedtypes.NamespaceKind {
```

Comment:
(No action required) Might make sense to add a helper for this pattern in a future PR.